packages/server-wallet/src/wallet/index.ts

```diff
@@ -421,6 +421,40 @@ export class Wallet extends EventEmitter<WalletEvent>
     return this.store.lockApp(channelId, criticalCode, handleMissingChannel);
   }
 
+  async closeChannels(channelIds: Bytes32[]): Promise<MultipleChannelOutput> {
+    for (const channelId of channelIds) {
+      const channel = await this.store.getChannel(channelId, undefined);
+
+      // TODO: (Objectives Rewrite) Keeping this here b/c we need to do input validation
+      // and check if its our turn and throw an error as existing tests expect
+      getOrThrow(CloseChannel.closeChannel(channel));
+
+      const objective: Objective = {
+        type: 'CloseChannel',
+        participants: [],
+        data: {targetChannelId: channelId, fundingStrategy: channel.fundingStrategy},
+      };
+
+      const objectiveToStore: ObjectiveStoredInDB = {
+        ...objective,
+        status: 'approved',
+        objectiveId: objectiveId(objective),
+      };
+
+      await ObjectiveModel.insert(objectiveToStore, this.knex);
```

Comment:
Should this use `this.store.addObjective`?

Reply:
Yes